### PR TITLE
[WIP] Move CI to GitLab

### PR DESCRIPTION
**Pull request by happenslol**
## Description

This PR moves automated testing and coverage reports to GitLab CI.

No changes to any engine code are made in this PR.

### CODE REVIEW

File: scripts/docs-ci-helper.bash [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      ;;
+    *)
+      exit 1
+  esac
+
+  # Write the newly built rev to the rev file
+  echo "$HEAD_REV" >> "$DIR/.rev"
+
+  # Write the invalidation path since we just rebuilt
+  echo "$INVALIDATION_PATH" >> ./book-paths-updated
```
Comment (azriel91) [inline scripts/docs-ci-helper.bash:53]: guessing `book-paths-updated` is cleared away on each new build?

File: scripts/docs-ci-helper.bash [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+function invalidate_aws {
+  # Check if there are any updated docs paths
+  if [[ -f "docs-paths-updated" ]]; then
```
Comment (azriel91) [inline scripts/docs-ci-helper.bash:58]: `docs-paths-updated` doesn't appear to be written to anywhere, is this left for later?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -96,15 +135,15 @@ book:
     - git lfs install
 
     # Build stable, master and wasm book
-    - ./.gitlab-ci-helper.sh build-book master book-public-master "/master/*"
-    - ./.gitlab-ci-helper.sh build-book $TAG_STABLE book-public-stable "/stable/*"
-    - ./.gitlab-ci-helper.sh build-book $BRANCH_WASM book-public-wasm "/wasm/*"
+    - ./scripts/.docs-ci-helper.bash build book master book-public-master "/master/*"
+    - ./scripts/.docs-ci-helper.bash build book $TAG_STABLE book-public-stable "/stable/*"
+    # - ./scripts/.docs-ci-helper.bash build book $BRANCH_WASM book-public-wasm "/wasm/*"
```
Comment (azriel91) [inline .gitlab-ci.yml:140]: yay placeholders! thanks!

File: scripts/docs-ci-helper.bash (hunk 2 above)
Comment (happenslol) [reply scripts/docs-ci-helper.bash:62]: Thanks, totally missed that!

**Comment by jojolepro** [comment]
Remove the WIP tag once ready for review :)

**Comment by jojolepro** [comment]
Closing due to inactivity, reopen if work resumes

**Pull request closed without merging**